erts/emulator/beam/jit/arm/instr_common.cpp

```diff
@@ -1079,12 +1079,8 @@ void BeamModuleAssembler::emit_i_is_tuple(const ArgVal &Fail,
     if (masked_types(Src, BEAM_TYPE_MASK_BOXED) == BEAM_TYPE_TUPLE) {
         comment("skipped header test since we know it's a tuple when boxed");
 
-        /* Note: ARG1 will NOT be set. This is OK since the operand
-         * for `current_tuple` has a type; that operand will not match
-         * the type-less operand for `get_tuple_element`. Thus, there
-         * will always be a `load_tuple_ptr` instruction emitted if
-         * this instruction is immediately followed by a
-         * `get_tuple_element` instruction. */
+        /* Must still set ARG1. */
+        emit_untag_ptr(ARG1, src.reg);
```

Comment:
This is identical on both branches, maybe hoist it?

Reply:
Will do.